homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -0,0 +1,359 @@
+            device = GeoRssServiceSensor(hass, category, data, name, icon,
+                                         unit_of_measurement)
+            devices.append(device)
+    async_add_devices(devices)
+
+    # Initialise update service.
+    updater = GeoRssServiceUpdater(hass, data, home_latitude, home_longitude,
+                                   url, radius_in_km, devices)
+    async_track_time_interval(hass, updater.async_update, scan_interval)
+    yield from updater.async_update()
+    return True
+
+
+class GeoRssServiceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, hass, category, data, name, icon,
+                 unit_of_measurement):
+        """Initialize the sensor."""
+        self.hass = hass
+        self._category = category
+        self._data = data
+        self._state = STATE_UNKNOWN
+        self._name = name
+        self._icon = icon
+        self._unit_of_measurement = unit_of_measurement
+        if category is not None:
+            id_base = category
+        else:
+            id_base = 'any'
+        if name is not None:
+            id_base = '{}_{}'.format(name, id_base)
+        self.entity_id = async_generate_entity_id(ENTITY_ID_FORMAT, id_base,
+                                                  hass=hass)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self._category is not None:
+            return self._category
+        else:
+            return 'Any'
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if isinstance(self._state, list):
+            return len(self._state)
+        else:
+            return self._state
+
+    @property
+    def should_poll(self):  # pylint: disable=no-self-use
+        """No polling needed."""
+        return False
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        matrix = {}
+        if self._state is not STATE_UNKNOWN:
+            for event in self._state:
+                matrix[event.title] = '{:.0f}km'.format(event.distance)
+        return matrix
+
+    def update(self):
+        """Update this sensor from the GeoRSS service."""
+        all_events = self._data.events
+        if self._category is None:
+            # Add all events regardless of category.
+            self._state = all_events
+        else:
+            # Group events by category.
+            my_events = []
+            for event in all_events:
+                if event.category == self._category:
+                    my_events.append(event)
+            self._state = my_events
+            _LOGGER.info("New state: %s", self._state)
```

Comment:
Unless there's extra info to provide, this isn't necessary. Hass core will already log state changes.